base/ltfinal.dtx

```diff
@@ -1051,28 +1055,35 @@
 %    mathematics, environment names, labels, etc.  A much better
 %    long-term solution is to use all-caps fonts, but these aren't
 %    generally available.
+%
+% Two wrappers around the L3 case changing functions.
+% |\protected| to make them mostly safe as replacemenst for |luppercase|
+% and |\lowercase|.  
 %    \begin{macrocode}
+\ExplSyntaxOn
+\protected\def\@text@lowercase{\text_lowercase:n}
+\protected\def\@text@uppercase{\text_uppercase:n}
```

Comment:
we have a different convention here `\@expl@...@@n` I suggest you follow that.

Reply:
yes I wondered about that but thought the expl ones were mostly more internal  but happy to change.